### PR TITLE
[node][edge][cli] upgrade Edge Runtime

### PR DESCRIPTION
**Pull request by Kikobeats**
We avoided to use `undici.WebSocket` because @jawj found a bug in the implementation.

The [fix was merged](https://github.com/nodejs/undici/pull/2106) in [undici@5.22.1](https://github.com/nodejs/undici/releases/tag/v5.22.1), so we can be back to use it.

The latest version of Edge Runtime exposes `undici.WebSocket` compatible with node.js 14, 16 & 18

### CODE REVIEW

File: packages/node/src/edge-functions/edge-handler.mts [hunk 1]
```diff
@@ -133,8 +133,18 @@ async function createEdgeRuntimeServer(params?: {
...
-    const WebSocket = (await import('ws')).WebSocket;
+    let WebSocket: any;
+
+    // undici's WebSocket handling is only available in Node.js >= 18
+    // so fallback to using ws for v16
+    if (Number(process.version.split('.')[0].substring(1)) < 18) {
+      // @ts-ignore
+      WebSocket = (await import('ws')).WebSocket;
+    } else {
+      WebSocket = (await import('undici')).WebSocket;
```
Comment (EndangeredMassa) [inline packages/node/src/edge-functions/edge-handler.mts]: **question (soft-blocking):** ‏Did you manually test this with projects that were failing [last time](https://github.com/vercel/vercel/pull/9893)?

**Review by EndangeredMassa — DISMISSED** [review]

**Comment by EndangeredMassa** [comment]
When I tried this localy, it failed with:

```
Error from API Route /api/edge-websocket: WebSocket is not defined
    at (api/edge-websocket.js)
```

I updated the `edge-runtime` and it worked: [`cd64eec` (#10051)](https://github.com/vercel/vercel/pull/10051/commits/cd64eec4e8069f7259512dc052f26f10a80a3f95)

**Review by EndangeredMassa — DISMISSED** [review]

**Review by EndangeredMassa — APPROVED** [review]

**Review by styfle — DISMISSED** [review]

File: .changeset/lucky-trainers-glow.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@vercel/node": patch
```
Comment (EndangeredMassa) [inline .changeset/lucky-trainers-glow.md:2]: **suggestion (blocking):** please ‏make sure there are lines for `cli` and `edge` packages as well

**Review by EndangeredMassa — CHANGES_REQUESTED** [review]

File: .changeset/lucky-trainers-glow.md (hunk 2 above)
Comment (Kikobeats) [reply .changeset/lucky-trainers-glow.md:2]: added although not sure why; changeset added what it detects.

File: .changeset/lucky-trainers-glow.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+"vercel": patch
+"@vercel/edge": patch
+---
+
+[node] use `undici.Websocket` when is possible
```
Comment (ecklf) [inline .changeset/lucky-trainers-glow.md:5]: Is this description accurate? Do we have a fallback for Node versions undici does not support?

**Review by ecklf — DISMISSED** [review]

File: .changeset/lucky-trainers-glow.md (hunk 3 above)
Comment (Kikobeats) [reply .changeset/lucky-trainers-glow.md:5]: good catch; fixed!

File: .changeset/lucky-trainers-glow.md (hunk 2 above)
Comment (EndangeredMassa) [reply .changeset/lucky-trainers-glow.md:2]: Sorry, I don't see the updated file. Was this updated to include `cli` and `edge`?

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -1,4 +1,8 @@
-lockfileVersion: '6.0'
```
Comment (Kikobeats) [inline pnpm-lock.yaml]: <picture data-single-emoji=":oh_no:" title=":oh_no:">:oh_no:</picture>

**Review by EndangeredMassa — DISMISSED** [review]
I updated the changelog file

**Review by cb1kenobi — DISMISSED** [review]
Tested creating a WebSocket and no issues.

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**